src/iface/ethernet.rs

```diff
@@ -644,3 +645,157 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::Packet;
+    use phy::{Loopback, ChecksumCapabilities};
+    use wire::{ArpOperation, ArpPacket, ArpRepr};
+    use wire::{EthernetAddress, EthernetFrame, EthernetProtocol};
+    use wire::{IpAddress, IpCidr, IpProtocol, IpRepr};
+    use wire::{Ipv4Address, Ipv4Repr};
+    use iface::{ArpCache, SliceArpCache, EthernetInterface};
+    use socket::SocketSet;
+
+    fn loopback_test<F>(f: F)
+            where F: FnOnce(&mut EthernetInterface<Loopback>) -> () {
+        // Create a basic device
+        let mut device = Loopback::new();
+
+        let mut arp_cache_entries: [_; 8] = Default::default();
```

Comment:
You can box everything here and then you can just return EthernetInterface by value.

Reply:
Good point. It's a bit more readable that way